packages/kbn-securitysolution-data-table/common/types/detail_panel.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+type EmptyObject = Record<string | number, never>;
```

Comment:
It might be worth keeping the detail panel as a separate package since it's used by the tables and session view, but @PhilippeOberti is working on the new version of this as well. Worth a discussion potentially